### PR TITLE
refactor: `TestScreen`

### LINKED ISSUE #310: Migrate `TestScreen` to TypeScript
- [x] Add correct types
- [x] Rewrite from Class to Functional component
- [x] Use Redux Hooks like `useDispatch` and `useSelector` instead of `connect`, `mapDispatchToProps` and `mapStateToProps`.

### PR DESCRIPTION
**Pull request by dansvend**
- Refactor `TestScreen` from React class component to functional component and hooks.

Closes #310

### CODE REVIEW

**Review by tormodAase — CHANGES_REQUESTED** [review]

File: hooks/useHearingTestProgressBar.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+
+import { selectTest } from "../store/test/reducer";
+
+export const useHearingTestProgressBar = () => {
```
[the remark this answers is not in the document]
Comment (dansvend) [reply hooks/useHearingTestProgressBar.ts]: Done in [b6d6057](https://github.com/equinor/hearing-test/pull/340/commits/b6d60575938c607db1c29cb02abdf0adc5715b83).

File: hooks/useHearingTestSoundFiles.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import { createSoundFile } from "../utils/sound/createSoundFile";
+
+export const useHearingTestSoundFiles = () => {
+  const silentSoundRef = useRef(
+    createSoundFile(require("../assets/audio/1000Hz_dobbel.wav"))
+  );
+  const demoModeSoundRef = useRef(
+    createSoundFile(require("../assets/audio/1000Hz_dobbel.wav"))
+  );
+  const soundsRef = useRef<{ [key: string]: Sound }>({});
```
[the remark this answers is not in the document]
Comment (dansvend) [reply hooks/useHearingTestSoundFiles.ts]: Done in [ae07a20](https://github.com/equinor/hearing-test/pull/340/commits/ae07a208e8919216a2c4271eb6d4b865dbf385f5).

File: hooks/useHearingTestSoundFiles.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  const [isSoundFilesLoaded, setIsSoundFilesLoaded] = useState(false);
+
+  const isDemoMode = getIsDemoModeEnabled();
```
[the remark this answers is not in the document]
Comment (dansvend) [reply hooks/useHearingTestSoundFiles.ts]: Done in [1ccf44d](https://github.com/equinor/hearing-test/pull/340/commits/1ccf44dcf22dc953fcb50a0c23f5060449c3c27e).

File: hooks/useHearingTestProgressBar.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    nodeCount -= subTests.length;
+    setNumberOfNodes(nodeCount);
+    setIsProgressBarInitialized(true);
+  };
+
+  useEffect(() => {
+    if (!isProgressBarInitialized && Object.keys(test).length > 0) {
+      initializeProgressBar();
+    }
+  }, [isProgressBarInitialized, test]);
```
[the remark this answers is not in the document]
Comment (dansvend) [reply hooks/useHearingTestProgressBar.ts]: Done in [ebc90f9](https://github.com/equinor/hearing-test/pull/340/commits/ebc90f9bb25fd9b36eba04728b83e682a17cbbc3).

File: contexts/VolumeContext.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+const useInitialSystemVolume = () => {
+  const [initialSystemVolume, setInitialSystemVolume] = useState(SYSTEM_VOLUME);
+
+  useEffect(() => {
+    const setInitialDeviceSystemVolume = async () => {
+      await SystemSetting.getVolume()
+        .then((volume) => setInitialSystemVolume(volume))
+        .catch((err) =>
+          console.error("Error setting initial system volume", err)
```
[the remark this answers is not in the document]
Comment (dansvend) [reply contexts/VolumeContext.tsx]: Done in [b7447c2](https://github.com/equinor/hearing-test/pull/340/commits/b7447c20908ee06d75ae723796917b79364fe547). We'll add toasts for rejected promises when we get a toast system implemented in [@equinor/mad-core](https://www.npmjs.com/package/@equinor/mad-core).

File: hooks/useHearingTestSoundFiles.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (dansvend) [reply hooks/useHearingTestSoundFiles.ts]: Done in [e25046d](https://github.com/equinor/hearing-test/pull/340/commits/e25046de60e26aa6b97f139766e96b33abf6a181).

**Review by tormodAase — APPROVED** [review]

**Review by tormodAase — APPROVED** [review]

**Pull request merged**